Makefile

```diff
@@ -1,114 +1,151 @@
-# Makefile to compile HMcode
-
-# Compiler
-FC = gfortran 
-
-# Standard flags
-FFLAGS = \
-	-fcheck=all \
-	-fmax-errors=4 \
-	-ffpe-trap=invalid,zero,overflow \
-	-fimplicit-none \
-	-O3 \
-	-std=gnu \
-	-ffree-line-length-none \
-	-fdefault-real-8 \
-	-fdefault-double-8 \
-	-lgfortran \
-	-lm
-
-# Extra debugging flags
-DEBUG_FLAGS = \
-	-Wall \
-	-fcheck=all \
-	-fbounds-check \
-	-fbacktrace \
-	-Og
-
-# Binary
-BIN = HMcode
-BIN_DEBUG = $(BIN)_debug
+ifeq (${FC},)
+FC = gfortran
+endif
+ifeq (${FC},f77)
+FC = gfortran
+endif
+
+UNAME = $(shell uname)
+ifeq (${UNAME}, Darwin)
+  LIBTOOL = libtool -static -o
+else
+  LIBTOOL = ar src
+endif
 
 # Source-code directory
-SRC_DIR = src
+LIB_SRC_DIR = library/src
+
+PYTHON_MOD = pyhmcode
+F90WRAP_SRC_DIR = f90wrap_helpers
+
 
 # Build directory
 BUILD_DIR = build
+F90WRAP_BUILD_DIR = f90wrap_pyhmcode
 
-# Fortran library directory (change this)
-MOD_DIR = library/src
+LIB_DIR = lib
+INCLUDE_DIR = include
 
 # Debug build directory
 DEBUG_BUILD_DIR = debug_build
 
-# Executable directory
-BIN_DIR = bin
+ifneq (${FC},gfortran)
+FFLAGS = -fpic -fpp -qopenmp -fp-model precise -W0 -WB -O3 -ipo -axCORE-AVX2 -fdefault-real-8 -fdefault-double-8 -ffpe-trap=invalid,zero,overflow
+DEBUGFLAGS = -fpp -qopenmp -g -check all -check noarg_temp_created -traceback -fpe0
+MODOUT = -module $(INCLUDE_DIR)
+DEBUG_MODOUT = -module $(INCLUDE_DIR)
+else
+FFLAGS =  -fPIC -cpp -fopenmp -O3 -ffree-line-length-none -std=gnu -fdefault-real-8 -fdefault-double-8
+DEBUGFLAGS = -g -O0 -fcheck=all -fbacktrace -cpp -fdec -fopenmp -ffree-line-length-none -Wall -fbounds-check -ffpe-trap=invalid,zero,overflow
+#add -fpe0 to check for floating point errors (think lowLike also throws these harmlessly)
+MODOUT = -J$(INCLUDE_DIR)
+DEBUG_MODOUT = -J$(INCLUDE_DIR)
+endif
 
 # Objects
 _OBJ = \
 	precision.o \
 	constants.o \
 	physics.o \
+	sorting.o \
 	basic_operations.o \
+	special_functions.o \
 	array_operations.o \
-	random_numbers.o \
 	file_info.o \
+	random_numbers.o \
 	table_integer.o \
-	special_functions.o \
 	interpolate.o \
 	solve_equations.o \
 	string_operations.o \
 	calculus_table.o \
-	camb_stuff.o \
-	sorting.o \
 	statistics.o \
 	calculus.o \
 	minimization.o \
+	camb_stuff.o \
 	cosmology_functions.o \
-	hmx.o 
+	hmx.o \
+	
```

Comment:
Was this re-ordering of the library files necessary? Just curious.

Reply:
I think so. This was created with trial and error. So it probably caused some problem until I fiddled it away. Can't really remember though.